docs/scanning-vendors.md

```diff
@@ -0,0 +1,457 @@
+
+The following sections in this article will cover the basic principles for how scanning vendors should use Red Hat security 
+data to accurately report on vulnerabilities, specifically for Red Hat containers images. 
+
+The examples used in this article will be for the following images: 
+
+[Repository: registry.redhat.io/rhel9/python-312 Tag: 1-25](https://catalog.redhat.com/software/containers/rhel9/python-312/657b088123df896ebfacf1f0?q=python&container-tabs=overview&image=66cf3054a2c0cf86bc022be9) 
+[Repository: registry.redhat.io/openshift4/ose-console-rhel9 Tag: v4.16.0-202409181705.p0.g0b1616c.assembly.stream.el9](https://catalog.redhat.com/software/containers/openshift4/ose-console-rhel9/65280984f0f695f11b13a24e?image=66eb1a1cdf6256d9be4690e6&architecture=amd64)
+
+## Package Identification and purls
+In order to accurately report on vulnerabilities in Red Hat products, scanning vendors must properly identify the 
+Red Hat package versions for RPMs, RPM modules and container first content, to identify any backported fixes. The
+following section provides guidance on how to find information about packages and how they are reported in Red Hat's 
+CSAF advisories and VEX  content. 
+
+### RPMs and RPM modules 
+An RPM package is a file format used by the Red Hat Package Manager (RPM) system for software distribution and management, 
+which package consists of an archive of files and metadata used to install and erase these files.
+
+There are two types of RPM packages: source RPMs and binary RPMs. Both types share the same file format and tooling, but have
+different contents and serve different purposes. A source RPM (SRPM) contains source code and a spec file, which 
+describes how to build the source code into a binary RPM, while a binary RPM contains the binaries built from the 
+sources and patches.
+
+Similarly, an RPM module is a set of RPM packages that represent a component and are usually installed together. A typical module 
+contains packages with an application, packages with the application-specific dependency libraries, packages with 
+documentation for the application, and packages with helper utilities. 
+
+Both binary RPMs and RPM modules installed in a container image can be discovered using the `rpm -qa` command.
+```
+# Example return of RPM query
+$ rpm -qa
+
+libgcc-11.3.1-4.3.el9.x86_64
+subscription-manager-rhsm-certificates-20220623-1.el9.noarch
+setup-2.13.7-9.el9.noarch
+filesystem-3.16-2.el9.x86_64
+basesystem-11-13.el9.noarch
+```
+
+Additionally, SRPMs can be discovered from a binary RPM by using the following command: 
+ ```
+# Example return of SRPM query
+$ rpm -q --qf "%{SOURCERPM}\n" libgcc-11.3.1-4.3.el9.x86_64    
+   
+gcc-11.3.1-4.3.el9.src.rpm
+ ```
+SRPMS, RPMs and RPM modules are represented in CSAF advisories and VEX data using the `rpm` purl type. 
+
+```
+# Example of a RPM purl
+pkg:rpm/redhat/libgcc-11.3.1-4.3.el9?arch=x86_64
+
+# Example of a SRPM purl 
+pkg:rpm/redhat/gcc?arch=src
```

Comment:
It is important to mention that such source rpm representation is when the particular package is associated with different than FIXED state. Once it's fixed, then additionally to the `arch=src` there is also version, like:  
`"purl": "pkg:rpm/redhat/ca-certificates@2024.2.69_v8.0.303-80.0.el8_10?arch=src"` 

What is more, the rpm modules are represented in a special way:  
`"purl": "pkg:rpm/redhat/python3x-pip@3.9?arch=src&rpmmod=python39:3.9"` 
Where we add additional `rpmmod`.

If we want to have examples here in this doc, let's have all possible examples. 
Or we can just have a link to the purl doc.

Reply:
Based on the purl guideline the `"purl": "pkg:rpm/redhat/python3x-pip@3.9?arch=src&rpmmod=python39:3.9"` shows a record for the `python3x-pip` RPM package version `3.9` which is part of the `python39` RPM module version also `3.9`. 

@mprpic can you confirm?